UserAgent/Views/Privacy Indicator/PrivacyIndicatorCanvasView.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+
+class PrivacyIndicatorCanvasView: UIView {
+    private var cachedSliceLayers: [CGColor: CAShapeLayer] = [:]
+    private var cachedStats: [WTMCategory: Int] = [:]
+    private var backgroundTrackLayer: CAShapeLayer?
+    private var greenLayer: CAShapeLayer?
+    private var strikeThroughLayer: CAShapeLayer?
+}
+
+extension PrivacyIndicatorCanvasView {
+    func render(with stats: TPPageStats, status: BlockerStatus) {
+        DispatchQueue.main.async { [weak self] in
+            if self == nil { return }
```

Comment:
As you don't need to force unwrap self in this scope so you don't need to check "self" is nil or not. The dispatch scope can be:
{ [weak self] in
     self?.cachedStats = WTMCategory.statsDict(from: stats)
     self?.render(status: status)
}

Reply:
@chrmod This is not blocking comment you can merge it.

Reply:
@shahenAnt make sense, thx